### PR TITLE
add decorator and policy to storage_files and propagate context for it

### PR DESCRIPTION
**Pull request by SuyogSoti**
For reviewers, there are two things to look for.

1. I add the `@distributed_trace` or `@distributed_trace_async` for every function that will lead to a network call.
1. I have added the DistributedTracingPolicy to the correct place.
1. I propagate the context with `tracing_context.with_current_context` any time a new thread is created. This should work with both `threading` and `ThreadPoolExecutor`. An example is the following:

`future = executor.submit(tracing_context.with_current_context(uploader.process_chunk), chunk)`

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file/azure/storage/file/file_client.py [hunk 1]
```diff
@@ -220,6 +221,7 @@ def from_connection_string(
         return cls(
             account_url, share=share, file_path=file_path, snapshot=snapshot, credential=credential, **kwargs)
 
+    @distributed_trace
     def generate_shared_access_signature(
```
Comment (annatisch) [inline sdk/storage/azure-storage-file/azure/storage/file/file_client.py]: No network calls in this function

File: sdk/storage/azure-storage-file/azure/storage/file/file_service_client.py [hunk 2]
```diff
@@ -132,6 +133,7 @@ def from_connection_string(
             kwargs['secondary_hostname'] = secondary
         return cls(account_url, credential=credential, **kwargs)
 
+    @distributed_trace
     def generate_shared_access_signature(
```
Comment (annatisch) [inline sdk/storage/azure-storage-file/azure/storage/file/file_service_client.py]: No network calls in this function

File: sdk/storage/azure-storage-file/azure/storage/file/share_client.py [hunk 3]
```diff
@@ -166,6 +166,7 @@ def from_connection_string(
         return cls(
             account_url, share=share, snapshot=snapshot, credential=credential, **kwargs)
 
+    @distributed_trace
     def generate_shared_access_signature(
```
Comment (annatisch) [inline sdk/storage/azure-storage-file/azure/storage/file/share_client.py]: No network calls in this function

File: sdk/storage/azure-storage-file/azure/storage/file/file_client.py (hunk 1 above)
Comment (SuyogSoti) [reply sdk/storage/azure-storage-file/azure/storage/file/file_client.py]: done

File: sdk/storage/azure-storage-file/azure/storage/file/file_service_client.py (hunk 2 above)
Comment (SuyogSoti) [reply sdk/storage/azure-storage-file/azure/storage/file/file_service_client.py]: done

File: sdk/storage/azure-storage-file/azure/storage/file/share_client.py (hunk 3 above)
Comment (SuyogSoti) [reply sdk/storage/azure-storage-file/azure/storage/file/share_client.py]: done

**Pull request merged**